--- origsrc/mate-image-viewer-1.2.0/doc/reference/Makefile.am	2012-03-07 09:10:13.000000000 -0600
+++ src/mate-image-viewer-1.2.0/doc/reference/Makefile.am	2012-04-16 21:29:00.277117300 -0500
@@ -93,19 +93,12 @@ INCLUDES= \
 	-I$(top_srcdir)/cut-n-paste/totem-screensaver		\
 	$(EOM_CFLAGS)
 
-if ENABLE_JPEG
-jpeg_LIB = $(top_builddir)/jpegutils/libeom-jpegutils.la
-endif
-
 GTKDOC_LIBS=\
 	$(top_builddir)/src/libeom.la 	\
-	$(top_builddir)/cut-n-paste/toolbar-editor/libtoolbareditor.la \
-	$(top_builddir)/cut-n-paste/totem-screensaver/libtotemscrsaver.la \
 	$(BINDING_LIBS)			\
 	$(LIBJPEG)			\
 	$(X_LIBS)			\
 	$(EOM_LIBS)			\
-	$(jpeg_LIB)			\
 	$(MORE_GTKDOC_LIBS)
 
 # This includes the standard gtk-doc make rules, copied by gtkdocize.
--- origsrc/mate-image-viewer-1.2.0/plugins/Makefile.am	2012-03-07 09:10:13.000000000 -0600
+++ src/mate-image-viewer-1.2.0/plugins/Makefile.am	2012-04-16 21:27:57.452524000 -0500
@@ -8,7 +8,7 @@ INCLUDES = \
 	$(WARN_CFLAGS)					\
 	-DEOM_LOCALEDIR=\""$(prefix)/$(DATADIRNAME)/locale"\"
 
-AM_LDFLAGS = -avoid-version -module
+AM_LDFLAGS = -avoid-version -module -no-undefined
 
 plugins_LTLIBRARIES = \
 	fullscreen/libfullscreen.la  \
@@ -23,14 +23,14 @@ fullscreen_libfullscreen_la_SOURCES = \
 	fullscreen/eom-fullscreen-plugin.h  \
 	fullscreen/eom-fullscreen-plugin.c
 
-fullscreen_libfullscreen_la_LIBADD = $(EOM_LIBS)
+fullscreen_libfullscreen_la_LIBADD = $(top_builddir)/src/libeom.la $(EOM_LIBS)
 
 # Reload plugin
 reload_libreload_la_SOURCES = \
 	reload/eom-reload-plugin.h  \
 	reload/eom-reload-plugin.c
 
-reload_libreload_la_LIBADD = $(EOM_LIBS)
+reload_libreload_la_LIBADD = $(top_builddir)/src/libeom.la $(EOM_LIBS)
 
 # Statusbar Date Plugin
 if HAVE_EXIF
@@ -40,7 +40,7 @@ statusbar_date_libstatusbar_date_la_SOUR
 	statusbar-date/eom-statusbar-date-plugin.h  \
 	statusbar-date/eom-statusbar-date-plugin.c
 
-statusbar_date_libstatusbar_date_la_LIBADD = $(EOM_LIBS)
+statusbar_date_libstatusbar_date_la_LIBADD = $(top_builddir)/src/libeom.la $(EOM_LIBS)
 plugins_in_files += statusbar-date/statusbar-date.eom-plugin.desktop.in
 endif
 
--- origsrc/mate-image-viewer-1.2.0/src/Makefile.am	2012-03-07 09:10:13.000000000 -0600
+++ src/mate-image-viewer-1.2.0/src/Makefile.am	2012-04-16 21:26:43.854314400 -0500
@@ -6,7 +6,7 @@ toolbar_LIB = $(top_builddir)/cut-n-past
 
 screensaver_LIB = $(top_builddir)/cut-n-paste/totem-screensaver/libtotemscrsaver.la
 
-noinst_LTLIBRARIES = libeom.la
+lib_LTLIBRARIES = libeom.la
 
 bin_PROGRAMS = eom
 
@@ -137,9 +137,9 @@ if HAVE_EXEMPI
 if !HAVE_EXIF
 INST_H_FILES +=				\
 	eom-exif-details.h
-endif !HAVE_EXIF
 libeom_la_SOURCES +=			\
 	eom-exif-details.c
+endif !HAVE_EXIF
 endif HAVE_EXEMPI
 
 libeom_la_CFLAGS =						\
@@ -174,7 +174,12 @@ libeom_la_CFLAGS += 			\
 	$(AM_CFLAGS)
 
 libeom_la_LIBADD += \
-	$(top_builddir)/bindings/python/eom.la
+	$(top_builddir)/bindings/python/eom.la \
+	$(toolbar_LIB)			\
+	$(screensaver_LIB)		\
+	$(jpeg_LIB)			\
+	$(LIBJPEG)			\
+	$(EOG_LIBS)
 endif
 
 libeom_la_LDFLAGS = -export-dynamic -no-undefined -export-symbols-regex "^[[^_]].*"
@@ -191,10 +196,7 @@ eom_CFLAGS = 							\
 eom_LDADD = 				\
 	libeom.la			\
 	$(EOM_LIBS)			\
-	$(LIBJPEG)			\
-	$(toolbar_LIB)			\
-	$(screensaver_LIB)		\
-	$(jpeg_LIB)
+	$(X_LIBS)
 
 eom_LDFLAGS = -export-dynamic -no-undefined -export-symbols-regex "^[[^_]].*"
 
