--- origsrc/mate-image-viewer-1.4.0/data/eom.pc.in	2012-07-12 13:38:32.000000000 -0500
+++ src/mate-image-viewer-1.4.0/data/eom.pc.in	2012-11-14 23:57:21.675752200 -0600
@@ -9,3 +9,4 @@ Description: The MATE Image Viewer
 Requires: gtk+-2.0, mateconf-2.0
 Version: @VERSION@
 Cflags: -I${includedir}
+Libs: -L${libdir} -leom
--- origsrc/mate-image-viewer-1.4.0/doc/reference/Makefile.am	2012-07-12 13:38:32.000000000 -0500
+++ src/mate-image-viewer-1.4.0/doc/reference/Makefile.am	2012-11-15 00:11:39.808912100 -0600
@@ -102,10 +102,10 @@ GTKDOC_LIBS=\
 	$(top_builddir)/cut-n-paste/toolbar-editor/libtoolbareditor.la \
 	$(top_builddir)/cut-n-paste/totem-screensaver/libtotemscrsaver.la \
 	$(BINDING_LIBS)			\
+	$(jpeg_LIB)			\
 	$(LIBJPEG)			\
 	$(X_LIBS)			\
 	$(EOM_LIBS)			\
-	$(jpeg_LIB)			\
 	$(MORE_GTKDOC_LIBS)
 
 # This includes the standard gtk-doc make rules, copied by gtkdocize.
--- origsrc/mate-image-viewer-1.4.0/plugins/Makefile.am	2012-07-12 13:38:32.000000000 -0500
+++ src/mate-image-viewer-1.4.0/plugins/Makefile.am	2012-11-14 23:59:24.245921300 -0600
@@ -8,7 +8,7 @@ INCLUDES = \
 	$(WARN_CFLAGS)					\
 	-DEOM_LOCALEDIR=\""$(prefix)/$(DATADIRNAME)/locale"\"
 
-AM_LDFLAGS = -avoid-version -module
+AM_LDFLAGS = -avoid-version -module -no-undefined -Wl,$(top_builddir)/src/libeom.exe.a
 
 plugins_LTLIBRARIES = \
 	fullscreen/libfullscreen.la  \
--- origsrc/mate-image-viewer-1.4.0/src/Makefile.am	2012-07-12 13:38:32.000000000 -0500
+++ src/mate-image-viewer-1.4.0/src/Makefile.am	2012-11-15 00:07:26.748474400 -0600
@@ -137,9 +137,9 @@ if HAVE_EXEMPI
 if !HAVE_EXIF
 INST_H_FILES +=				\
 	eom-exif-details.h
-endif !HAVE_EXIF
 libeom_la_SOURCES +=			\
 	eom-exif-details.c
+endif !HAVE_EXIF
 endif HAVE_EXEMPI
 
 libeom_la_CFLAGS =						\
@@ -190,13 +190,14 @@ eom_CFLAGS = 							\
 
 eom_LDADD = 				\
 	libeom.la			\
-	$(EOM_LIBS)			\
-	$(LIBJPEG)			\
 	$(toolbar_LIB)			\
 	$(screensaver_LIB)		\
-	$(jpeg_LIB)
+	$(jpeg_LIB)			\
+	$(LIBJPEG)			\
+	$(EOM_LIBS)
 
 eom_LDFLAGS = -export-dynamic -no-undefined -export-symbols-regex "^[[^_]].*"
+eom_LDFLAGS += -Wl,--out-implib,libeom.exe.a
 
 BUILT_SOURCES = 			\
 	eom-enum-types.c		\
@@ -233,6 +234,10 @@ endif
 
 CLEANFILES = $(BUILT_SOURCES)
 
+install-data-local:
+	$(mkinstalldirs) $(DESTDIR)$(libdir)
+	$(INSTALL_DATA) libeom.exe.a $(DESTDIR)$(libdir)/libeom.dll.a
+
 dist-hook:
 	cd $(distdir); rm -f $(BUILT_SOURCES)
 
